File: bank-service/src/main/java/rs/edu/raf/dto/CardDTO.java
package rs.edu.raf.dto;

import java.math.BigDecimal;
public record CardDTO(
    Long id,
    String number,
    String type,
    CardIssuerDTO cardIssuer,
    String name,
    Long creationDate,
    Long expirationDate,
    BankAccountDTO bankAccount,
    int cvv,
    BigDecimal cardLimit,
    String status
) { }

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CardDTO.kt
package org.example.bankservicekotlin.dto

import java.math.BigDecimal

data class CardDTO(
    val id: Long?,
    val number: String,
    val type: String,
    val cardIssuer: CardIssuerDTO,
    val name: String,
    val creationDate: Long,
    val expirationDate: Long,
    val bankAccount: BankAccountDTO,
    val cvv: Int,
    val cardLimit: BigDecimal,
    val status: String
)
